pkg/kubectl/cmd/attach.go

```diff
@@ -167,7 +167,7 @@ func (p *AttachOptions) Run() error {
 		if err != nil {
 			return err
 		}
-		if pod.Status.Phase != api.PodRunning {
+		if pod.Status.Phase != api.PodRunning && (pod.Status.Phase == api.PodPending && len(pod.Spec.InitContainers) < 1) {
```

Comment:
ad. 1. no problem, I can change it

ad. 2. It will fail with `error: error executing remote command: error attaching to container: container not found ("<CONTAINTER_NAME>")`. Is that OK?